src/vote_signer_proxy.rs

```diff
@@ -26,27 +28,65 @@ pub enum VoteError {
     LeaderInfoNotFound,
 }
 
-pub struct VoteSignerProxy {
+pub struct RemoteVoteSigner {
     rpc_client: RpcClient,
+}
+
+impl RemoteVoteSigner {
+    pub fn new(signer: SocketAddr) -> Self {
+        Self {
+            rpc_client: RpcClient::new_from_socket(signer),
+        }
+    }
+}
+
+impl VoteSigner for RemoteVoteSigner {
+    fn register(
+        &self,
+        pubkey: Pubkey,
+        sig: &Signature,
+        msg: &[u8],
+    ) -> jsonrpc_core::Result<Pubkey> {
+        let params = json!([pubkey, sig, msg]);
+        let resp = RpcRequest::RegisterNode
+            .retry_make_rpc_request(&self.rpc_client, 1, Some(params), 5)
+            .unwrap();
+        let vote_account: Pubkey = serde_json::from_value(resp).unwrap();
+        Ok(vote_account)
+    }
+    fn sign(&self, pubkey: Pubkey, sig: &Signature, msg: &[u8]) -> jsonrpc_core::Result<Signature> {
+        let params = json!([pubkey, sig, msg]);
+        let resp = RpcRequest::SignVote
+            .make_rpc_request(&self.rpc_client, 1, Some(params))
+            .unwrap();
+        let vote_signature: Signature = serde_json::from_value(resp).unwrap();
+        Ok(vote_signature)
+    }
+    fn deregister(&self, pubkey: Pubkey, sig: &Signature, msg: &[u8]) -> jsonrpc_core::Result<()> {
+        let params = json!([pubkey, sig, msg]);
+        let _resp = RpcRequest::DeregisterNode
+            .retry_make_rpc_request(&self.rpc_client, 1, Some(params), 5)
+            .unwrap();
+        Ok(())
+    }
+}
+
+pub struct VoteSignerProxy {
```

Comment:
```rust
pub struct VoteSignerProxy<T: VoteSigner + Send + Sync> {
    signer: T,
    ...
}
```

Reply:
VoteSignerProxy object is passed to fullnode::new() and later stored in pub struct fullnode.
Wouldn't struct fullnode require trait <T: VoteSigner> as well?

Reply:
Yeah, I'm afraid those parameters are a bit viral like that.  It seems pretty awful to add a type parameter for every object we mock in Fullnode, but let's do it anyway. I think it's the lesser of two evils.  Maybe when we get better with one of those mocking libraries, we can find a way to get those out. In the meantime, we could always create a type alias that fills in all those types for our unit tests.

Reply:
Yeah, I'm afraid those parameters are a bit viral like that. It seems pretty awful to add a type parameter for every object we mock in Fullnode, but let's do it anyway. I think it's the lesser of two evils. Maybe when we get better with one of those mocking libraries, we can find a way to get those out. In the meantime, we could always create a type alias that fills in all those types for our unit tests.